cmd/s2i/main.go

```diff
@@ -165,7 +165,7 @@ $ s2i build . centos/ruby-22-centos7 hello-world-app
 	cmdutil.AddCommonFlags(buildCmd, cfg)
 
 	buildCmd.Flags().BoolVar(&(cfg.RunImage), "run", false, "Run resulting image as part of invocation of this command")
-	buildCmd.Flags().BoolVar(&(cfg.DisableRecursive), "recursive", true, "Fetch all git submodules when cloning application repository")
+	buildCmd.Flags().BoolVar(&(cfg.IgnoreSubmodules), "ignore-submodules", false, "Ignore all git submodules when cloning application repository")
```

Comment:
(I really don't want to retain deprecated backwards compatibility for something that has confusing behavior.  better to just get people to move to the right flag)